### PR TITLE
Change from boost to std regex

### PR DESCRIPTION
**Pull request by ntonci**

### CODE REVIEW

**Comment by HannesSommer** [comment]
The difficult thing here is that you're regular expressions were interpreted with the perl dialect and now they are going to be interpreted with the ECMAScript dialect (default in c++11). I believe that both dialects agree on those two. 
Assuming that it LGTM.

UNLESS you use the GNU libstdc++ library version 4.8, which is fundamentally broken in its regex support (see https://github.com/ethz-asl/Schweizer-Messer/pull/184). And that is why jenkins fails here. Is it an option to drop trusty as target platform? Otherwise I can only think of the hybrid approach I've decided painfully for in the PR https://github.com/ethz-asl/Schweizer-Messer/pull/184. Use either boost or c++11 regex depending on the library version :).

File: cv_bridge/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 2.8)
 project(cv_bridge)
 
+add_definitions(-std=c++11)
```
Comment (HannesSommer) [inline cv_bridge/CMakeLists.txt]: Actually I would to this differently. E.g. like this: https://github.com/ethz-asl/oomact/blob/master/oomact/CMakeLists.txt#L9-L14 
(never mind the `-std=c++11` in the line above the ones I've marked. It is not necessary for this library)

**Review by HannesSommer — COMMENTED** [review]
See my other comment about trusty.

**Comment by HannesSommer** [comment]
Wait a minute: the branch you merge into is called `kinetic`. Could it be that this refers to ROS kinetic? Why do you even let Jenkins build it on trusty with indigo then? @mfehr ?

**Comment by mfehr** [comment]
@HannesSommer this repo is a dependecy of maplab (as a submodule of [thirdparty_submodules](https://github.com/ethz-asl/thirdparty_submodules/)), which is deployed and tested on OSX, 14.04 and 16.04. If I remember correctly providing this repo as a dependency (instead of using the native ROS version) was a prerequisite of upgrading to a newer opencv version. That's why we added jenkins builds for all three operating systems.

**Comment by HannesSommer** [comment]
Oh my, then it should probably be working on all those systems. I'd suggest to use the approach from: https://github.com/ethz-asl/Schweizer-Messer/pull/184 then. It is compiled for with the same family. It would be even simpler for this repo, because the regular expression itself does not need to be adapted depending on boost or c++11. And as soon as trusty is not supported any longer the cleanup is simple.

Otherwise we need to fix the linking issue itself. @ntonci Which change actually triggered the problem this PR is trying to fix? Or was it just a special computer that was failing?

**Comment by ntonci** [comment]
@HannesSommer Couple of packages on my computer that are depending on this one were failing because of boost::regex

**Comment by HannesSommer** [comment]
Okay, then at least two computers have this type of problem (assuming the Linux laptop on which hand eye calibration should be performed was not your computer ..). And if I remember right it wasn't an easy one.
If nobody has a problem with the suggest hybrid approach (like in SM), it is probably the easiest option.

File: cv_bridge/src/cv_bridge.cpp [hunk 2]
```diff
@@ -72,6 +78,22 @@ static int depthStrToInt(const std::string depth) {
 
 int getCvType(const std::string& encoding)
 {
+
+#ifndef USE_BOOST_REGEX
+  std::regex FormatterFormatGroupPattern("\\$\\{([a-zA-Z]+)[}]", std::regex::egrep);
```
Comment (HannesSommer) [inline cv_bridge/src/cv_bridge.cpp]: You can drop both versions of `FormatterFormatGroupPattern`. And my comment about the `[}]`. Not needed in your case.

File: cv_bridge/src/cv_bridge.cpp [hunk 3]
```diff
@@ -72,6 +78,22 @@ static int depthStrToInt(const std::string depth) {
...
+  // The [}] is a workaround for a bug in at least libstdc++ 5.4 that throws on "\\}" (BUT not on "\\{").
+  using std::match_results;
+  using std::regex_search;
```
Comment (HannesSommer) [inline cv_bridge/src/cv_bridge.cpp]: I don't think you need `regex_search`.

**Review by HannesSommer — CHANGES_REQUESTED** [review]
LGTM otherwise.

File: cv_bridge/CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,12 @@
 cmake_minimum_required(VERSION 2.8)
 project(cv_bridge)
 
+if(CMAKE_VERSION VERSION_LESS "3.1")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std+c++11")
```
Comment (HannesSommer) [inline cv_bridge/CMakeLists.txt]: Here is a typo: that is Jenkins problem. ..

Comment (ntonci) [reply cv_bridge/CMakeLists.txt]: Sorry for that, surprisingly it was still building on my computer :/

File: cv_bridge/src/cv_bridge.cpp (hunk 3 above)
Comment (ntonci) [reply cv_bridge/src/cv_bridge.cpp]: @HannesSommer Actually do I even need match_results and should I add regex_match? It is still compiling without however

**Review by HannesSommer — APPROVED** [review]

File: cv_bridge/CMakeLists.txt (hunk 4 above)
Comment (HannesSommer) [reply cv_bridge/CMakeLists.txt]: Your `cmake` is apparently new enough :).

**Pull request merged**